rust/connlib/tunnel/src/lib.rs

```diff
@@ -433,23 +450,17 @@ where
         packet: MutableIpPacket,
         dest: IpAddr,
         peer: &ConnectedPeer<TRoleState::Id>,
+        peer_queue: &tokio::sync::mpsc::Sender<Bytes>,
     ) {
         let peer_id = peer.inner.conn_id;
 
-        match peer.inner.encapsulate(packet, dest, write_buf) {
+        match peer.inner.encapsulate(packet, write_buf) {
             Ok(None) => {}
             Ok(Some(b)) => {
-                tokio::spawn({
-                    let channel = peer.channel.clone();
-                    let mut sender = self.stop_peer_command_sender.clone();
-
-                    async move {
-                        if let Err(e) = channel.write(&b).await {
-                            tracing::error!(resource_address = %dest, err = ?e, "failed to handle packet {e:#}");
-                            let _ = sender.send(peer_id).await;
-                        }
-                    }
-                });
+                tracing::trace!(target: "wire", action = "writing", to = "peer", %dest);
+                if peer_queue.try_send(b).is_err() {
```

Comment:
We should differentiate between `Full` and `Disconnected` here. In case of `Full`, we can just drop the packet, but if we are disconnected, we should clean up the connection.

Reply:
I was in the process of moving everything towards just cleaning up the connection when wireguard session expire.

Since we're no longer using sctp if the connection fails we can't distinguish that other than using keep alive messages and since we want to get rid of webrtc anyways we shouldn't be depending on webrtc's keep alives.

I will work on a PR next that restarts ice with each keep alive tiemout in wireguard(keeping the older connection alive in case it's a fluke), fail the connection only if the new ice restart fails.

Reply:
Sounds good!

Reply:
> I will work on a PR next that restarts ice with each keep alive tiemout in wireguard(keeping the older connection alive in case it's a fluke), fail the connection only if the new ice restart fails.

Shall we delay that for now? I think this will be easier to implement once we have `str0m`. I'd like to get rid of all `async` functions on the tunnel and remove all locks first. Those are quite the productivity hit I find :(

Reply:
I'm fine delaying it for now but I might need to implement some preliminary version for beta (better re-connectivity is key for UX!)